websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,279 @@
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?watched/threads")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?finde-new/")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?forums/")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = `Browsing throguht ${forumTitle.toLowerCase()}`;
+					presenceData.state = `"${forumTitle}"`;
+					presenceData.buttons = [
+						{
+							label: "View Forum",
+							url: location.href
+						}
+					];
+				} else if (search.startsWith("?threads/")) {
+					presenceData.largeImageKey = "reading";
+					presenceData.details = "Reading a thread";
+					presenceData.state = `"${forumTitle}"`;
+					presenceData.buttons = [
+						{
+							label: "Read Thread",
+							url: location.href
+						}
+					];
+				} else if (search.startsWith("?members/")) {
+					if (document.title.startsWith("Notable Members")) {
+						presenceData.largeImageKey = "search";
+						presenceData.details = "Browsing notable members";
+					} else {
+						presenceData.largeImageKey = "reading";
+						presenceData.details = "Viewing a member's page";
+						presenceData.state = `@${forumTitle}`;
+						presenceData.buttons = [
+							{
+								label: "View Member",
+								url: location.href
+							}
+						];
+					}
+				} else if (search.startsWith("?search/")) {
+					presenceData.largeImageKey = "search";
+					if (document.title.startsWith("Error" || "Search"))
+						presenceData.details = "Searching for something...";
+					else {
+						presenceData.details = `Searching for ${
+							document.title
+								.split("Search Results for Query: ")[1]
+								.split(" | Sonic and Sega Retro Forums")[0]
+						}`;
+					}
+				} else if (search.startsWith("?conversations/")) {
+					presenceData.largeImageKey = "logo_1024";
+					presenceData.details = "Viewing an unsupported page";
+				} else if (search.startsWith("?account/")) {
+					presenceData.largeImageKey = "logo_1024";
+					presenceData.details = "Viewing an unsupported page";
+				} else if (search.startsWith("?help/")) {
+					presenceData.largeImageKey = "logo_1024";
+					presenceData.details = "Viewing an unsupported page";
+				} else if (
+					location.href ===
+					"https://forums.sonicretro.org/index.php?forums/-/index.rss"
+				) {
+					presenceData.largeImageKey = "logo_1024";
+					presenceData.details = "Viewing an unsupported page";
+				}
+			} else if (pathname === "/index.php" && !search) {
+				presenceData.largeImageKey = "forums";
+				presenceData.details = "Zooming through the main page";
+			} else {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
```

Comment:
all of these conditions except for the second to last one are doing the exact same, you can probably improve this